### ISSUE TITLE
Crash when adding last card to arena deck

### ISSUE THREAD

**Issue by HandEyeProtege** [Epix37/Hearthstone-Deck-Tracker#1510]
I got a crash report when creating an arena deck, when I selected the last card for the deck.  Below is the contents of the crash report.  (I trimmed a bunch of the older log statements because github wouldn't let me submit the whole thing.)

If it's helpful, the choices were Doomguard / Argent Watchman / Void Terror; I chose Doomguard.

########## 9/22/2015 12:10:53 PM ##########
System.NullReferenceException: Object reference not set to an instance of an object.
   at Hearthstone_Deck_Tracker.Hearthstone.GameV2.<NewArenaCard>b__20(Deck d)
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   at Hearthstone_Deck_Tracker.Hearthstone.GameV2.<NewArenaCard>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<ThrowAsync>b__4(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)

[...I trimmed a bunch of the log statements here...]

[12:00:20 PM] GameEventHandler: --- Opponent turn 6 ---
[12:00:20 PM] Opponent Draw: id=59, cardId=, cardName=, turn=6
[12:00:30 PM] GameEventHandler: --- Game was won! ---
[12:00:30 PM] Gamestats: Current Game ended after 6 turns
[12:00:30 PM] GameStats: Assigned current game to deck: Fel Mech Mage
[12:00:30 PM] GameEventHandler: waiting for game mode detection
[12:00:39 PM] RachelleHandler: Updating GoldProgress, waiting for GameModeDetection...
[12:00:42 PM] GameEventHandler: set rank to 13
[12:00:43 PM] Game: set CurrentGameMode to Casual
[12:00:43 PM] : Updated GoldProgress from 1/3 - 0/100 to 1/3 - 0/100 for region US.
[12:00:43 PM] GameEventHandler: waiting for ranked detection
[12:00:43 PM] Game: set CurrentGameMode to Ranked
[12:00:43 PM] Game: set CurrentGameMode to Ranked
[12:00:43 PM] GameEventHandler: waiting for game mode to be saved to game
[12:00:44 PM] GameEventHandler: Set CurrentGameStats.GameMode to Ranked
[12:00:44 PM] GameEventHandler: Saving DeckStats
[12:00:44 PM] HearthStatsManager: trying to upload match: Win vs Hunter, 9/22/2015 11:56:05 AM
[12:00:44 PM] HearthStatsManager: background process indicator ON
[12:00:44 PM] HearthStatsAPI: uploading match: Win vs Hunter, 9/22/2015 11:56:05 AM
[12:00:44 PM] HearthStatsAPI: assigned id to match: 30064510
[12:00:44 PM] HearthStatsManager: background process indicator OFF
[12:00:44 PM] HearthStatsManager: success uploading match Win vs Hunter, 9/22/2015 11:56:05 AM
[12:00:49 PM] Game: set CurrentGameMode to Arena
[12:00:57 PM] Game: set CurrentGameMode to Arena
[12:01:00 PM] : Created new arena deck: Warlock
[12:01:30 PM] UpdateOverlay: Refreshed overlay topmost status
[12:01:31 PM] : Added new card to arena deck: EX1_170 (1/30)
[12:01:47 PM] UpdateOverlay: Refreshed overlay topmost status
[12:01:47 PM] : Added new card to arena deck: EX1_302 (2/30)
[12:01:59 PM] UpdateOverlay: Refreshed overlay topmost status
[12:02:00 PM] : Added new card to arena deck: EX1_019 (3/30)
[12:02:12 PM] UpdateOverlay: Refreshed overlay topmost status
[12:02:12 PM] : Added new card to arena deck: AT_101 (4/30)
[12:02:56 PM] UpdateOverlay: Refreshed overlay topmost status
[12:02:57 PM] : Added new card to arena deck: BRM_020 (5/30)
[12:03:11 PM] UpdateOverlay: Refreshed overlay topmost status
[12:03:11 PM] : Added new card to arena deck: CS2_222 (6/30)
[12:03:24 PM] UpdateOverlay: Refreshed overlay topmost status
[12:03:25 PM] : Added new card to arena deck: AT_094 (7/30)
[12:03:52 PM] UpdateOverlay: Refreshed overlay topmost status
[12:03:52 PM] : Added new card to arena deck: EX1_020 (8/30)
[12:04:17 PM] UpdateOverlay: Refreshed overlay topmost status
[12:04:18 PM] : Added new card to arena deck: CS2_117 (9/30)
[12:04:28 PM] UpdateOverlay: Refreshed overlay topmost status
[12:04:28 PM] : Added new card to arena deck: EX1_303 (10/30)
[12:04:38 PM] UpdateOverlay: Refreshed overlay topmost status
[12:04:38 PM] : Added new card to arena deck: CS2_179 (11/30)
[12:05:11 PM] UpdateOverlay: Refreshed overlay topmost status
[12:05:11 PM] : Added new card to arena deck: GVG_006 (12/30)
[12:05:22 PM] UpdateOverlay: Refreshed overlay topmost status
[12:05:23 PM] : Added new card to arena deck: GVG_096 (13/30)
[12:05:59 PM] UpdateOverlay: Refreshed overlay topmost status
[12:06:00 PM] : Added new card to arena deck: CS2_059 (14/30)
[12:06:31 PM] UpdateOverlay: Refreshed overlay topmost status
[12:06:32 PM] : Added new card to arena deck: CS2_062 (15/30)
[12:06:42 PM] UpdateOverlay: Refreshed overlay topmost status
[12:06:42 PM] : Added new card to arena deck: EX1_316 (16/30)
[12:06:53 PM] UpdateOverlay: Refreshed overlay topmost status
[12:06:54 PM] : Added new card to arena deck: NEW1_026 (17/30)
[12:07:06 PM] UpdateOverlay: Refreshed overlay topmost status
[12:07:06 PM] : Added new card to arena deck: EX1_319 (18/30)
[12:07:19 PM] UpdateOverlay: Refreshed overlay topmost status
[12:07:19 PM] : Added new card to arena deck: NEW1_020 (19/30)
[12:07:33 PM] UpdateOverlay: Refreshed overlay topmost status
[12:07:33 PM] : Added new card to arena deck: EX1_309 (20/30)
[12:08:07 PM] UpdateOverlay: Refreshed overlay topmost status
[12:08:08 PM] : Added new card to arena deck: GVG_006 (21/30)
[12:08:17 PM] UpdateOverlay: Refreshed overlay topmost status
[12:08:18 PM] : Added new card to arena deck: FP1_012 (22/30)
[12:08:35 PM] UpdateOverlay: Refreshed overlay topmost status
[12:08:36 PM] : Added new card to arena deck: CS2_057 (23/30)
[12:08:47 PM] UpdateOverlay: Refreshed overlay topmost status
[12:08:48 PM] : Added new card to arena deck: GVG_015 (24/30)
[12:08:58 PM] UpdateOverlay: Refreshed overlay topmost status
[12:08:58 PM] : Added new card to arena deck: EX1_046 (25/30)
[12:09:17 PM] UpdateOverlay: Refreshed overlay topmost status
[12:09:18 PM] : Added new card to arena deck: EX1_162 (26/30)
[12:09:50 PM] UpdateOverlay: Refreshed overlay topmost status
[12:09:57 PM] UpdateOverlay: Refreshed overlay topmost status
[12:09:57 PM] : Added new card to arena deck: EX1_302 (27/30)
[12:10:31 PM] UpdateOverlay: Refreshed overlay topmost status
[12:10:31 PM] : Added new card to arena deck: AT_020 (28/30)
[12:10:40 PM] UpdateOverlay: Refreshed overlay topmost status
[12:10:41 PM] : Added new card to arena deck: EX1_048 (29/30)
[12:10:52 PM] UpdateOverlay: Refreshed overlay topmost status

**Comment by Epix37** [comment]
No idea what's causing this but it should at least no longer crash now.

**Comment by GaryKing** [comment]
I haven't had a crash when adding Arena games, but I've had problems adding Arena games nonetheless, lately.

I've got the setting to "auto-import Arena games", so when I choose the 30th card in an Arena game, it should automatically import in HDT. This works about 75% of the time. The remaining 25%, HDT will, for some reason, not recognize one or two cards, which are usually the 3rd copy of a specific card. So for instance, if I drafted 3 Flamestrikes, then HDT sometimes won't auto-import the game. So when I go to manually import it, it will indeed say "29 / 30 cards", and then I notice that Flamestrike only says "2" in HDT when it should be 3, so I fix that manually.

Hopefully this latest change fixes that too.
